src/main/java/io/split/android/client/localhost/LocalhostSplitClient.java

```diff
@@ -143,22 +144,70 @@ public Map<String, SplitResult> getTreatmentsWithConfig(List<String> featureFlag
 
     @Override
     public Map<String, String> getTreatmentsByFlagSet(@NonNull String flagSet, @Nullable Map<String, Object> attributes) {
-        return null;
+        try {
+            return mTreatmentManager.getTreatmentsByFlagSet(flagSet, attributes, mIsClientDestroyed);
+        } catch (Exception exception) {
+            Logger.e(exception);
+
+            Map<String, String> result = new HashMap<>();
+            Set<String> namesByFlagSets = mSplitsStorage.getNamesByFlagSets(Collections.singletonList(flagSet));
+            for (String featureFlagName : namesByFlagSets) {
+                result.put(featureFlagName, Treatments.CONTROL);
+            }
+
+            return result;
+        }
     }
 
     @Override
     public Map<String, String> getTreatmentsByFlagSets(@NonNull List<String> flagSets, @Nullable Map<String, Object> attributes) {
-        return null;
+        try {
+            return mTreatmentManager.getTreatmentsByFlagSets(flagSets, attributes, mIsClientDestroyed);
+        } catch (Exception exception) {
```

Comment:
It seems that there is room to reuse the code inside the `catch` branch of the `getTreatments[WithConfig]ByFlagSet[s]` methods. Maybe you can move it into a private method, and call it with `flagSets` or `Collections.singletonList(flagSet)` as an argument depending on the case.